node/network/statement-distribution/src/vstaging/tests/cluster.rs

```diff
@@ -1213,4 +1259,216 @@ fn cluster_messages_imported_after_new_leaf_importable_check() {
+			)
+			.await;
+			connect_peer(
+				&mut overseer,
+				peer_b.clone(),
+				Some(vec![state.discovery_id(v_b)].into_iter().collect()),
+			)
+			.await;
+			connect_peer(
+				&mut overseer,
+				peer_c.clone(),
+				Some(vec![state.discovery_id(v_c)].into_iter().collect()),
+			)
+			.await;
+
+			send_peer_view_change(&mut overseer, peer_a.clone(), view![relay_parent]).await;
+			send_peer_view_change(&mut overseer, peer_b.clone(), view![relay_parent]).await;
+			send_peer_view_change(&mut overseer, peer_c.clone(), view![relay_parent]).await;
+		}
+
+		activate_leaf(&mut overseer, local_para, &test_leaf, &state, true).await;
+
+		answer_expected_hypothetical_depth_request(
+			&mut overseer,
+			vec![],
+			Some(relay_parent),
+			false,
+		)
+		.await;
+
+		// Confirm the candidate locally so that we don't send out requests.
+		{
+			let validator_index = state.local.as_ref().unwrap().validator_index;
+			let statement = state
+				.sign_full_statement(
+					validator_index,
+					Statement::Seconded(candidate),
+					&SigningContext { parent_hash: relay_parent, session_index: 1 },
+					pvd,
+				)
+				.clone();
+
+			share_local_statement(&mut overseer, relay_parent, statement).await;
+
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages::NetworkBridgeTx(NetworkBridgeTxMessage::SendValidationMessage(peers, _)) if peers == vec![peer_a, peer_b, peer_c]
+			);
+
+			answer_expected_hypothetical_depth_request(
+				&mut overseer,
+				vec![],
+				// TODO: Is this right?
+				None,
+				false,
+			)
+			.await;
+		}
+
+		// Send statement from peer A.
+		{
+			println!("statement from A");
+			let statement = state
+				.sign_statement(
+					v_a,
+					CompactStatement::Seconded(candidate_hash),
+					&SigningContext { parent_hash: relay_parent, session_index: 1 },
+				)
+				.as_unchecked()
+				.clone();
+
+			send_peer_message(
+				&mut overseer,
+				peer_a.clone(),
+				protocol_vstaging::StatementDistributionMessage::Statement(relay_parent, statement),
+			)
+			.await;
+
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages::NetworkBridgeTx(NetworkBridgeTxMessage::ReportPeer(p, r))
+					if p == peer_a && r == BENEFIT_VALID_STATEMENT_FIRST => { }
+			);
+
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages::NetworkBridgeTx(NetworkBridgeTxMessage::SendValidationMessage(peers, _)) if peers == vec![peer_b, peer_c]
+			);
+		}
+
+		// Send statement from peer B.
+		{
+			println!("statement from B");
+			let statement = state
+				.sign_statement(
+					v_b,
+					CompactStatement::Seconded(candidate_hash),
+					&SigningContext { parent_hash: relay_parent, session_index: 1 },
+				)
+				.as_unchecked()
+				.clone();
+
+			send_peer_message(
+				&mut overseer,
+				peer_b.clone(),
+				protocol_vstaging::StatementDistributionMessage::Statement(relay_parent, statement),
+			)
+			.await;
+
+			// TODO: Why is this sent out twice?
```

Comment:
Looks like the first time the subsystem got a Seconded statement from v_b, right?

Reply:
Think I don't understand what `BENEFIT_VALID_STATEMENT_FIRST` is for. I was under the impression that it's a reward given to the first peer that advertises a given candidate. The doc is not really helpful: `Peer was the first to provide a valid statement`.

Reply:
The reward is for the first peer to provide a valid statement. Since the two statements are different `Seconded` by `v_b` and `Seconded` by `v_a`, we give out the valid statement first reward for each.